SYCL/ESIMD/api/functional/ctors/common.hpp

```diff
@@ -14,36 +14,26 @@
 #pragma once
 
 #include "../common.hpp"
+#include <string>
 
 namespace esimd_test::api::functional::ctors {
 
-template <typename DataT, int NumElems, typename ContextT>
+template <int NumElems, typename ContextT>
 class TestDescription : public ITestDescription {
 public:
-  TestDescription(size_t index, DataT retrieved_val, DataT expected_val,
-                  const std::string &data_type)
-      : m_data_type(data_type), m_retrieved_val(retrieved_val),
-        m_expected_val(expected_val), m_index(index) {}
+  TestDescription(const std::string &data_type) : m_data_type(data_type) {}
```

Comment:
Why you do not provide a backward compatibility? Without the backward compatibility new tests will be failed on CI when this PR will be merged

Reply:
Would you mind to take a look on a backward compatibility suggestion? This PR removes unused `DataT` template parameter, so we have a difference in template parameters also.